src/table_compaction.cc

```diff
@@ -114,7 +114,12 @@ Status TableMgr::compactLevelItr(const CompactOptions& options,
     SizedBuf max_key_table;
     SizedBuf::Holder h_max_key_table(max_key_table);
     if (max_table) {
-        max_table->file->getMaxKey(max_key_table);
+        if(max_table->file->isEmpty()){
+            _log_info( myLog, "Table is empty, setting table's min key as the max key");
+            max_table->minKey.copyTo(max_key_table);
+        } else {
+            max_table->file->getMaxKey(max_key_table);
```

Comment:
To be safe, can we do `getMaxKey` first and then copy `minKey` to `max_key_table` if `max_key_table` is empty? I know `isEmpty` will be `true` when `getMaxKey` returns empty, but just in case and also considering any future change.

You can keep `isEmpty` API, I think that will be useful.

Reply:
Ok. But here we are assuming if `max_key_table` is empty that means table is empty. But are there any other cases where it can be return empty?

Reply:
That's correct in the current code. But the concern is that if `max_key_table` is changed in the future that returns an empty key under some other situation, but this compaction code is not properly updated so that it proceeds with the empty key. I want to avoid all kinds of such situations that are hard to predict.

Reply:
Sure. Done.